lib/posix/posix.nim

```diff
@@ -586,6 +586,8 @@ proc fstatvfs*(a1: cint, a2: var Statvfs): cint {.
   importc, header: "<sys/statvfs.h>".}
 
 proc chmod*(a1: cstring, a2: Mode): cint {.importc, header: "<sys/stat.h>", sideEffect.}
+when hasLchmod:
```

Comment:
simplify:
* remove all `hasLchmod` instances:
* use this:
```nim
when defined(osx) or defined(freebsd):
  proc lchmod*(a1: cstring, a2: Mode): cint {.importc, header: "<sys/stat.h>", sideEffect.}
```
* and then in os.nim use `when declared(lchmod)`